azure-functions-java-sdktypes/src/main/java/com/microsoft/azure/functions/sdktype/blob/BlobContainerHydrator.java

```diff
@@ -0,0 +1,136 @@
+package com.microsoft.azure.functions.sdktype.blob;
+
+import com.microsoft.azure.functions.sdktype.SdkTypeHydrator;
+
+import java.lang.reflect.Method;
+import java.util.logging.Logger;
+
+/**
+ * Reflection logic for building a BlobContainerClient from BlobContainerMetaData,
+ * supporting both connection string usage and managed identity usage.
+ */
+public class BlobContainerHydrator implements SdkTypeHydrator<BlobContainerMetaData> {
+    private static final Logger LOGGER = Logger.getLogger(BlobContainerHydrator.class.getName());
+
+    @Override
+    public Object createInstance(BlobContainerMetaData metaData) throws Exception {
+        String containerName = metaData.getContainerName();
+        String envVar = metaData.getConnectionEnvVar();
+
+        LOGGER.info("BlobContainerHydrator: Using environment variable name as prefix: " + envVar);
+
+        // Check if environment variable => connection string
+        String maybeConnString = System.getenv(envVar);
+        if (maybeConnString != null && isConnectionString(maybeConnString)) {
+            LOGGER.info("Detected connection string usage from environment variable: " + envVar);
+            return buildWithConnectionString(containerName, maybeConnString);
+        } else {
+            // interpret envVar as prefix for managed identity
+            LOGGER.info("Detected managed identity usage. Prefix: " + envVar);
+
+            String accountName = System.getenv(envVar + "__accountName");
+            String serviceUri = System.getenv(envVar + "__serviceUri");
+            String blobServiceUri = System.getenv(envVar + "__blobServiceUri");
+            String clientId = System.getenv(envVar + "__clientId");
+
+            String endpoint = resolveEndpoint(accountName, serviceUri, blobServiceUri);
+            Object credential = buildManagedIdentityCredential(clientId);
+
+            return buildWithManagedIdentity(containerName, endpoint, credential);
+        }
+    }
+
+    private boolean isConnectionString(String val) {
+        return val.contains("AccountKey=") || val.contains("DefaultEndpointsProtocol=");
+    }
+
+    private Object buildWithConnectionString(String containerName, String connStr) throws Exception {
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        Class<?> builderClass = cl.loadClass("com.azure.storage.blob.BlobContainerClientBuilder");
+        Object builder = builderClass.getDeclaredConstructor().newInstance();
+
```

Comment:
To improve the code, we can use the Adapter/Builder Pattern. This allows us to simplify the reflection-based code needed to build the BlobClient. 

I think having something as follow would be much cleaner

````
public class BlobClientBuilderAdapter {
    
    private final Object builderInstance;
    private final Class<?> builderClass;

    public BlobClientBuilderAdapter(ClassLoader classLoader) throws Exception {
        // Load the builder class using reflection

    public BlobClientBuilderAdapter withConnectionString(String connectionString) throws Exception {
    .......
        return this;
    }


similarly withconnectionString((String connectionString)

and so on

```